src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
+                print("Error while fetching Model Details for Model card.", e)
+                return ModelDetails()
+        
+        def update_dataset_details(exp_config: dict) -> DatasetDetails:
+            """
+            Function to initialize DatasetDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (DatasetDetails): initialized dataset details object.
+            """
+            try:
+                dataset_splits = {"split_config": exp_config["dataset_iterators"]["config"]["split_configs"], 
+                              "splits_percentage": exp_config["splitted_dataset_iterators"]["config"]["split_configs"]}
+            
+                return DatasetDetails(considered_dataset = exp_config["dataset_iterators"]["config"]["dataset_identifier"], dataset_splits = dataset_splits)
+            except Exception as e:
+                print("Error while fetching Dataset Details for Model card.", e)
+                return DatasetDetails()
+        
+        def update_training_details(exp_config: dict, gs_config: dict) -> TrainingDetails:
+            """
+            Function to initialize TrainingDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (TrainingDetails): initialized training details object.
+            """
+            try:
+                hyperparams = {}
+                hyperparams["optimizer"] = exp_config["optimizer"]["config"]["params"]
+                return TrainingDetails( hyperparams = hyperparams, loss_func = exp_config["train_component"]["config"]["loss_fun_config"]["tag"], optimizer = exp_config["optimizer"]["config"]["optimizer_key"])
+            except Exception as e:
+                print("Error while fetching Training Details for Model card.", e)
+                return TrainingDetails()
+        
+        def update_evaluation_details(exp_config: dict) -> EvalDetails:
+            """
+            Function to initialize EvalDetails object.
+            :params:
+                    exp_config (dict): Experiment configuration.
+            :returns:
+                    obj (EvalDetails): initialized evaluation details object.
+            """
+            loss_funcs = []
+            metrics = []
+            try:
+                for loss_func_config in exp_config["eval_component"]["config"]["loss_funs_config"]:
+                    loss_funcs.append(loss_func_config["tag"])
+            
+                for metric_config in exp_config["eval_component"]["config"]["metrics_config"]:
+                    metrics.append({"name": metric_config["tag"], "params": metric_config["params"]})
+
+                return EvalDetails(loss_funcs = loss_funcs, metrics = metrics)
+            except Exception as e:
+                print("Error while fetching Eval Details for Model card.", e)
```

Comment:
we need improved exception handling. see above.

Reply:
done